### PR TITLE
DT-235 Implement Update Living Unit service

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/java/net/syscon/elite/service/MovementUpdateService.java [hunk 1]
```diff
@@ -27,19 +27,17 @@ public MovementUpdateService(ReferenceDomainService referenceDomainService, Loca
     public OffenderSummary moveToCell(final Long bookingId, final Long livingUnitId, final String reasonCode, final LocalDateTime dateTime) {
         referenceDomainService.getReferenceCodeByDomainAndCode(CELL_MOVE_REASON.getDomain(), reasonCode, false);
         final var offenderSummary = getOffenderSummary(bookingId);
-        final var location = locationService.getLocation(livingUnitId);
 
-        if (!offenderSummary.getAgencyLocationId().equals(location.getAgencyId())) {
```
Comment (mikehalmamoj) [inline src/main/java/net/syscon/elite/service/MovementUpdateService.java]: Move agency check down to updateLivingUnitId service.

File: src/main/java/net/syscon/elite/service/MovementUpdateService.java [hunk 2]
```diff
@@ -27,19 +27,17 @@ public MovementUpdateService(ReferenceDomainService referenceDomainService, Loca
...
-            throw new IllegalArgumentException(format("Move to living unit in prison %s invalid for offender in prison %s", location.getAgencyId(), offenderSummary.getAgencyLocationId()));
+        if (offenderSummary.getInternalLocationId().equals(String.valueOf(livingUnitId))) {
```
Comment (mikehalmamoj) [inline src/main/java/net/syscon/elite/service/MovementUpdateService.java]: Introduced skip of update if not changed.

File: src/main/java/net/syscon/elite/service/BookingService.java [hunk 3]
```diff
@@ -634,7 +666,16 @@ public MilitaryRecords getMilitaryRecords(final Long bookingId) {
 
     @Transactional
     public void updateLivingUnit(final Long bookingId, final Long livingUnitId) {
-        // TODO DT-235 implement this
+        final var location = locationService.getLocation(livingUnitId);
+        var offenderBooking = offenderBookingRepository.findById(bookingId)
+                .orElseThrow(new EntityNotFoundException(format("Offender booking for booking id %d not found", bookingId)));
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/service/BookingService.java]: There are some static factory methods available on the `EntityNotFoundException` class...meh.

File: src/test/java/net/syscon/elite/repository/jpa/repository/OffenderBookingRepositoryTest.java [hunk 4]
```diff
@@ -261,6 +274,19 @@ private OffenderCourtCase offenderCourtCase(final Long caseIdentifier) {
                 .agencyLocation(agencyLocationRepository.findById("COURT1").orElseThrow())
                 .build();
     }
+
+    @Test
+    void updateLivingUnitId() {
+        var offenderBooking = repository.findById(-1L).orElseThrow();
+        offenderBooking.setLivingUnitId(22L);
+        repository.save(offenderBooking);
+        TestTransaction.flagForCommit();
```
Comment (matt-crowson-moj) [inline src/test/java/net/syscon/elite/repository/jpa/repository/OffenderBookingRepositoryTest.java]: Can just do `entityManager.flush()` here if you want instead...meh.

File: src/main/java/net/syscon/elite/repository/jpa/model/OffenderBooking.java [hunk 5]
```diff
@@ -60,6 +59,9 @@
     @JoinColumn(name = "OFFENDER_ID", nullable = false)
     private Offender offender;
 
+    @Column(name = "LIVING_UNIT_ID")
+    private Long livingUnitId;
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/repository/jpa/model/OffenderBooking.java]: Do we not have an Domain object we can map this to instead of just Long?

**Pull request merged**